### PR TITLE
Campaign cost calculator

### PR DESCRIPTION
**Pull request by johntwyman**
## Description

Adds a Service Manager that calculates inbound, outbound and total SMS costs for active or archived campaigns.

It has specific requirements to be function, most notably the inclusion of the PL/v8 language extension for PostgreSQL.

All requirements and installation instructions are noted in a README.md file.

The extension is released under the MIT license.

### CODE REVIEW

**Comment by johntwyman** [comment]
Ok @anthonyblond @peterkmurphy the PR is ready for review.

File: src/extensions/service-managers/campaign-cost-calculator/README.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+This service manager calculates the SMS costs for started (and archived) campaigns.
+
+Importantly, it calculates the cost on the basis of the number of message
+_segments_ used. This ensures a more accurate (though not 100% foolproof) estimate
+of costs.
+
+## Requirements
+
+- [PostgreSQL](https://www.postgresql.org/)
+- [PL/v8 language extension for PostgreSQL](https://plv8.github.io/)
```
Comment (anthonyblond) [inline src/extensions/service-managers/campaign-cost-calculator/README.md:12]: I don't think this is the case any more?

**Review by peterkmurphy — APPROVED** [review]

**Review by peterkmurphy — COMMENTED** [review]
Line 26 of the README.md has 'pricea'. Is that meant to be 'price' or 'prices'?

File: src/extensions/service-managers/campaign-cost-calculator/README.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+You will need administrator privileges in your database to perform installation.
+
+## Installation
+
+### Modify your .env file
+
+1. Add `campaign-cost-calculator` to your `SERVICE_MANAGERS` variable
+2. Add `CAMPAIGN_COST_CURRENCY` and assign it a three letter currency code in quotes (eg. `CAMPAIGN_COST_CURRENCY='USD'`)
+3. Add `CAMPAIGN_COST_INBOUND` and assign it the number (eg. `CAMPAIGN_COST_INBOUND=0.0025`)
+4. Add `CAMPAIGN_COST_OUTBOUND` and assign it the outbound pricea (eg. `CAMPAIGN_COST_OUTBOUND=0.055`)
```
Comment (anthonyblond) [inline src/extensions/service-managers/campaign-cost-calculator/README.md:25]: sp

**Review by anthonyblond — APPROVED** [review]

**Review by peterkmurphy — APPROVED** [review]
Apart from the 'pricea" part, it looks functional and attractive. I approve.

**Pull request merged**